scripts/ethereum-dev-node.ts

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env -S yarn node -r ts-node/register/transpile-only
+
+import * as fs from "fs/promises";
+import execa from "execa";
+import { deployAll } from "radicle-contracts";
+import * as ethers from "ethers";
+import waitOn from "wait-on";
+
+const ethAccountFile = "sandbox/.local-eth-account";
+
+main().catch(e => {
+  console.error(e);
+  process.exit(1);
+});
+
+async function main() {
+  let devEthAccount;
+  try {
+    devEthAccount = (await fs.readFile(ethAccountFile, "utf-8")).trim();
+  } catch (err) {
+    throw new Error(
+      `Failed to read address of development account from ${ethAccountFile}:\n  ${err.message}`
+    );
+  }
+
+  const ganache = execa(
+    "yarn",
+    [
+      "run",
+      "ganache-cli",
+      "--mnemonic",
+      "image napkin cruise dentist name plunge crisp muscle nest floor vessel blush",
+      "--defaultBalanceEther",
+      "1000",
+    ],
+    { stdio: "inherit" }
+  );
+  await waitOn({
+    resources: ["tcp:127.0.0.1:8545"],
+    timeout: 10000,
+  });
+
+  const provider = new ethers.providers.JsonRpcProvider(
+    "http://localhost:8545"
+  );
+  const signer = provider.getSigner(0);
+  const txCount = await signer.getTransactionCount();
+  if (txCount !== 0) {
+    throw new Error(
+      "Deployer account has non-zero transaction count. You need to reset your chain"
```

Comment:
Is it possible for Genache to randomly load an old state on startup? If so, maybe we should always reset it first?

Reply:
Good point. Ganache uses persistent state if the `--db` flag is provided. We don’t do this so we can remove this check.